File: 73-set-matrix-zeroes/set-matrix-zeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        unordered_set<int> row, col;
        int r = matrix.size(), c = matrix[0].size();

        for(int i = 0; i < r; i++) {
            for(int j = 0; j < c; j++) {
                if(matrix[i][j] == 0) {
                    row.insert(i);
                    col.insert(j);
                }
            }
        } 
        for(int x: row) {
            for(int j = 0; j < c; j++) matrix[x][j] = 0;
        }

        for(int x: col) {
            for(int i = 0; i < r; i++) matrix[i][x] = 0;
        }
    }
};

File: 73-set-matrix-zeroes/set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        row = set()
        col = set()
        r = len(matrix)
        c = len(matrix[0])

        for i in range(0, r):
            for j in range(0, c):
                if matrix[i][j] == 0:
                    row.add(i)
                    col.add(j)

        for i in row: 
            for j in range(0, c):
                matrix[i][j] = 0

        for j in col:
            for i in range(0, r):
                matrix[i][j] = 0


        